google/cloud/storage/async/client.cc

```diff
@@ -38,6 +38,56 @@ AsyncClient::AsyncClient(Options options) {
 AsyncClient::AsyncClient(std::shared_ptr<AsyncConnection> connection)
     : connection_(std::move(connection)) {}
 
+std::pair<AsyncRewriter, AsyncToken> AsyncClient::StartRewrite(
+    BucketName const& source_bucket, std::string source_object_name,
+    BucketName const& destination_bucket, std::string destination_object_name,
+    Options opts) {
+  return StartRewrite(source_bucket, std::move(source_object_name),
+                      destination_bucket, std::move(destination_object_name),
+                      google::storage::v2::RewriteObjectRequest{},
+                      std::move(opts));
+}
+
+std::pair<AsyncRewriter, AsyncToken> AsyncClient::StartRewrite(
+    BucketName const& source_bucket, std::string source_object_name,
+    BucketName const& destination_bucket, std::string destination_object_name,
+    google::storage::v2::RewriteObjectRequest request, Options opts) {
```

Comment:
I feel like this overload is asking for trouble... The application could supply different values in `source_bucket` and `request.source_bucket()`, e.g..

@coryan - Was this intentional? Do we really want this overload, when the other two exist?

Reply:
It was intentional, but you make a good argument.  I will send a PR to remove it and avoid these in the future.